hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/LogCleaner.java

```diff
@@ -86,8 +87,9 @@ public LogCleaner(final int period, final Stoppable stopper, Configuration conf,
 
   @Override
   protected boolean validate(Path file) {
-    return AbstractFSWALProvider.validateWALFilename(file.getName())
-        || MasterProcedureUtil.validateProcedureWALFilename(file.getName());
+    return AbstractFSWALProvider.validateWALFilename(file.getName()) ||
+      MasterProcedureUtil.validateProcedureWALFilename(file.getName()) ||
+      file.getName().endsWith(LocalStore.ARCHIVED_WAL_SUFFIX);
   }
 
   @Override
```

Comment:
Man, we should have called this file the WALCleaner, not LogCleaner. Not your fault.

Reply:
Not sure if it is safe to change the name. Can be another issue?

Reply:
Yes. Of course.